source/NPC.cpp

```diff
@@ -151,6 +152,16 @@ void NPC::Load(const DataNode &node)
 			else
 				child.PrintTrace("Skipping unrecognized attribute:");
 		}
+		else if(child.Token(0) == "on" && child.Size() >= 2)
+		{
+			int eventType = ShipEvent::TypeFromString(child.Token(1));
```

Comment:
we def should be using this whole method at the start of this function as well, within a lambda but that could be for another PR

Reply:
The beginning of `Load` does a very different thing to this.

Reply:
Oh right its parsing die save etc? Why would we use different names here?

Reply:
Supporting actions triggered by individual ship events is fairly simple, more complex behaviours like hte objectives is more complicated.
What would it mean to have an `on save` NPC action? Complete this mission with the whole NPC alive? Does the NPC get told when we reach the destination? Is some event sent at that time?
Potential for extension there, but it seems excessive, and not necessary if actions can be triggered by ship events already.